### PR TITLE
PLAT-140541: Migrate moonstone pattern samples

### PR DESCRIPTION
**Pull request by juwonjeong**
Enact-DCO-1.0-Signed-off-by: Juwon Jeong (juwon.jeong@lge.com)

### Checklist
* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Put moonstone related pattern samples into moonstone folder

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Update dependencies 
Fix pattern-locale-switching samples 
Fix code based on enact code style 

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
PLAT-140541

### Comments

### CODE REVIEW

File: moonstone/pattern-dynamic-panel/src/App/App.js [hunk 1]
```diff
@@ -1,7 +1,6 @@
-import Changeable from '@enact/ui/Changeable';
 import kind from '@enact/core/kind';
 import MoonstoneDecorator from '@enact/moonstone/MoonstoneDecorator';
-import React from 'react';
+import Changeable from '@enact/ui/Changeable';
```
Comment (jeonghee27) [inline moonstone/pattern-dynamic-panel/src/App/App.js:3]: @MikyungKim  I'm not sure if the order is correct or not.

File: moonstone/pattern-activity-panels/src/App/App.js [hunk 2]
```diff
@@ -1,25 +1,16 @@
...
-	static defaultProps = {
-		index: 0
-	};
-
+class App extends Component {
 	constructor (props) {
 		super(props);
 		this.state = {
-			index: this.props.index
+			index: 0
```
Comment (jeonghee27) [inline moonstone/pattern-activity-panels/src/App/App.js:13]: @juwonjeong Could you explain why do change this part?

File: moonstone/pattern-dynamic-panel/src/components/FileBrowser/FileBrowser.js [hunk 3]
```diff
@@ -96,7 +97,7 @@ const FileBrowserBase = kind({
 				<Image src={filePhotos[leaf.replace('.jpg', '')]} />;
 
 			return (
-				<DynamicPanel path={path} {...rest}>
+				<DynamicPanel {...rest} path={path}>
```
Comment (jeonghee27) [inline moonstone/pattern-dynamic-panel/src/components/FileBrowser/FileBrowser.js:100]: If there is no any specific reason, rest should be on the last to override props.

File: moonstone/pattern-locale-switching/src/components/LocaleSwitch/LocaleSwitch.js [hunk 4]
```diff
@@ -1,14 +1,15 @@
 import Button from '@enact/moonstone/Button';
-import {contextTypes} from '@enact/i18n/I18nDecorator';
-import {connect} from 'react-redux';
 import Input from '@enact/moonstone/Input';
+import {I18nContextDecorator} from '@enact/i18n/I18nDecorator';
```
Comment (jeonghee27) [inline moonstone/pattern-locale-switching/src/components/LocaleSwitch/LocaleSwitch.js:3]: Please check order. i / m

File: moonstone/pattern-locale-switching/src/components/LocaleSwitch/LocaleSwitch.js [hunk 5]
```diff
@@ -45,6 +46,9 @@ class LocaleSwitch extends Component {
 	}
 }
 
-LocaleSwitch.contextTypes = contextTypes;
+const LocaleSwitch = I18nContextDecorator(
```
Comment (jeonghee27) [inline moonstone/pattern-locale-switching/src/components/LocaleSwitch/LocaleSwitch.js:49]: Please make sure this part has changed as intended. @MikyungKim

File: moonstone/pattern-virtualgridlist-api/src/main.js [hunk 6]
```diff
@@ -1,5 +1,4 @@
 import {Provider} from 'react-redux';
```
Comment (jeonghee27) [inline moonstone/pattern-virtualgridlist-api/src/main.js:1]: Could you run this app and try to scroll?

File: moonstone/pattern-activity-panels-deep-linking/src/App/App.js [hunk 7]
```diff
@@ -27,10 +26,10 @@ const App = kind({
 	render: ({onFourthPanel, onNavigate, onSecondPanel, onThirdPanel, path, ...rest}) => {
 		return (
 			<RoutablePanels {...rest} onSelectBreadcrumb={onNavigate} path={path}>
```
Comment (jeonghee27) [inline moonstone/pattern-activity-panels-deep-linking/src/App/App.js:28]: Don't we need to default path? when i open the app, the second page is shown.

File: moonstone/pattern-locale-switching/src/components/LocaleSwitch/LocaleSwitch.js (hunk 4 above)
Comment (juwonjeong) [reply moonstone/pattern-locale-switching/src/components/LocaleSwitch/LocaleSwitch.js]: It's fixed.

File: moonstone/pattern-virtualgridlist-api/src/main.js (hunk 6 above)
Comment (juwonjeong) [reply moonstone/pattern-virtualgridlist-api/src/main.js:1]: It's fixed.

File: moonstone/pattern-activity-panels-deep-linking/src/App/App.js (hunk 7 above)
Comment (juwonjeong) [reply moonstone/pattern-activity-panels-deep-linking/src/App/App.js:28]: I think it is intended to open with second page as mentioned at the end of [README.md](https://github.com/enactjs/samples/tree/master/pattern-activity-panels-deep-linking) file.

**Review by jeonghee27 — APPROVED** [review]
LGTM

File: moonstone/pattern-locale-switching/src/components/LocaleSwitch/LocaleSwitch.js (hunk 5 above)
Comment (0x64) [reply moonstone/pattern-locale-switching/src/components/LocaleSwitch/LocaleSwitch.js:49]: This change is from hook migration of I18nDecorator. Ref: enactjs/enact/pull/2727

**Pull request merged**